ref_arch/multi_tenant.rst

```diff
@@ -0,0 +1,393 @@
+
+  SELECT name, cost_model, state, monthly_budget
+    FROM campaigns
+   WHERE company_id = 5
+  ORDER BY monthly_budget DESC
+  LIMIT 10;
+
+This is a typical query for a multi-tenant application because it restricts the results to data from a single company. Each tenant, in this case an advertising company, will be accessing only their own data.
+
+Any column of the :code:`campaigns` table could be its distribution column, but let's compare how this query performs for either of two options: :code:`id` and :code:`company_id`.
+
+DIAGRAM: show id pulling from all workers, and company_id routed to one
+
+If we distribute by the campaign id, then campaign shards will be spread across multiple workers irrespective of company. Finding the top ten monthly campaign budgets for a company requires asking all workers for their local top ten and doing a final sort and filter on the coordinator. If we distribute by :code:`company_id`, on the other hand, then Citus can detect by the presence of :code:`WHERE company_id = 5` that all relevant information will be on a single worker. Citus can route the entire query to that worker for execution and pass the results through verbatim.
+
+The order/limit query slightly favors distribution by :code:`company_id`. However JOIN queries differ more dramatically.
+
+.. note::
+
+  In our normalized schema above, the ads table does not have a company_id column because it can retrieve that information through the campaigns table. If we want to distribute the ads table by company id however, we would need to denormalize the schema slightly and add that column. The query below assumes we have done this, and we'll talk more about this technique later.
+
+::
+
+  -- running campaigns which receive the most clicks and impressions
+  -- for a single tenant
+
+  SELECT campaigns.id, campaigns.name, campaigns.monthly_budget,
+         sum(impressions_count) as total_impressions,
+         sum(clicks_count) as total_clicks
+  FROM ads, campaigns
+  WHERE ads.company_id = campaigns.company_id
+  AND campaigns.company_id = 5
+  AND campaigns.state = 'running'
+  GROUP BY campaigns.id, campaigns.name, campaigns.monthly_budget
+  ORDER BY total_impressions, total_clicks;
+
+DIAGRAM: show id repartitioning, and company_id routing
+
+For this query, distributing by campaign id is quite bad. Workers must use a lot of network traffic to pull related information together for the join, in a process called *repartitioning.* Routing the query for execution in a single worker avoids the overhead, and is possible when distributing by :code:`company_id`. The placement of related information together on a worker is called *co-location.*
+
+These queries indicate a general design pattern: distributing shards by tenant id (such as the company id) allows Citus to route queries to individual workers for efficient processing. This fits multi-tenant applications which join structured information together per-tenant.
```

Comment:
Also, other than joins, I'd explain the benefits of sharding by tenant for transactions and foreign keys. We probably don't need to do diagrams etc but a description of why that's advisable and how it helps would be useful.